tapcfg/server.go

```diff
@@ -275,7 +276,13 @@ func genServerConfig(cfg *Config, cfgLogger btclog.Logger,
 		SyncBatchSize:       defaultUniverseSyncBatchSize,
 	})
 
-	runtimeID := prand.Int63() // nolint:gosec
+	var runtimeIDBytes [8]byte
```

Comment:
Any reason to not increase to 32 bytes so we don't have to think about it?

Reply:
Could do both but p sure root cause was the RNG.

Reply:
> Any reason to not increase to 32 bytes so we don't have to think about it?

Yeah, just to make things non-breaking on the RPC interface.

I think 8 bytes should be waaaay enough for this. I'm pretty sure this was a combination of weak RNG+docker+macOS that lead to duplicates. I tried to reproduce this on my machine with a dummy docker setup and wasn't able to.

Using the better RNG should definitely make this almost impossible to hit.

Unfortunately neither a unit nor an integration test can help here, since as long as things are in the same process, we can never get a collision as even the weak RNG is a package-level variable. So if code from the same process request randomness, it will be different for each call as the calls are serialized. This can only be reproduced by spawning multiple processes within a very short time (which isn't super easy to do with our current setup).

I'm voting to just leave things the way they are and rely on the CPRNG being much better for this.